mysql-test/suite/rpl/t/rpl_add_pk_on_extra_columns_on_replica.test

```diff
@@ -0,0 +1,176 @@
+# ==== Purpose ====
+#
+# This test verifies that update and delete events on a table are properly
+# applied by the replication applier threads when replica table has extra
+# columns, has primary key defined on any of the extra columns.
+#
+# ==== References ====
+#
+# PS-7578: Replication failure with UPDATE when replica server has a PK and
+#          source not
+
+--source include/have_binlog_format_row.inc
+--source include/master-slave.inc
+
+--source include/rpl_connection_slave.inc
+SET @saved_slave_rows_search_algorithms= @@global.slave_rows_search_algorithms;
+
+# Create an empty helper file which will be later used to load to the server.
+--let $LOAD_FILE= $MYSQLTEST_VARDIR/tmp/query_map.dat
+--write_file $LOAD_FILE
+EOF
+
+################################################################################
+# Here we write queries into the above file in the format
+#
+#          "query_identifier|query"
+#
+# so that this file can be used to load the queries into the table and fetch
+# the queries based on the identifier and execute it on the server for testing.
+#
+# Example: "CREATE_0|CREATE TABLE t1 (field1 char(15));"
+#          query_identifier = CREATE_0
+#          query = CREATE TABLE t1 (field1 char(15));
+#
+################################################################################
+
+############################
+# Variants of CREATE TABLE #
+############################
+# 1. Table with no keys.
+--exec echo "CREATE_0|CREATE TABLE t1 (field1 char(15));" >> $LOAD_FILE
+# 2. Table with a key.
+--exec echo "CREATE_1|CREATE TABLE t1 (field1 CHAR(15), field2 INT DEFAULT 1010, KEY idx (field1));" >> $LOAD_FILE
+# 3. Table with a composite key.
+--exec echo "CREATE_2|CREATE TABLE t1 (field1 CHAR(15), field2 INT DEFAULT 1010, KEY idx (field1, field2));" >> $LOAD_FILE
+
+########################
+# Query to insert data #
+########################
+--exec echo "DATA|INSERT INTO t1(field1) VALUES ('row1'); INSERT INTO t1(field1) VALUES ('row2'); INSERT INTO t1(field1) VALUES ('row3');" >> $LOAD_FILE
+
+###########################
+# Variants of ALTER TABLE #
+###########################
+# 1. No PK.
+--exec echo "ALTER_0|ALTER TABLE t1 ADD COLUMN extra_col INT NOT NULL DEFAULT 1;" >> $LOAD_FILE
+# 2. PK with AUTO_INCREMENT on additional column (adding one column).
+--exec echo "ALTER_1|ALTER TABLE t1 ADD COLUMN extra_col INT NOT NULL AUTO_INCREMENT PRIMARY KEY;" >> $LOAD_FILE
+# 3. Composite PK on both source's and additional columns.
+--exec echo "ALTER_2|ALTER TABLE t1 ADD COLUMN extra_col INT NOT NULL DEFAULT 1, ADD PRIMARY KEY (field1, extra_col);" >> $LOAD_FILE
+# 4. PK with AUTO_INCREMENT on additional column (adding two columns).
+--exec echo "ALTER_3|ALTER TABLE t1 ADD COLUMN extra_col1 INT NOT NULL DEFAULT 100, ADD COLUMN extra_col2 INT NOT NULL AUTO_INCREMENT, ADD PRIMARY KEY (extra_col2);" >> $LOAD_FILE
+
```

Comment:
can you add a column at the beginning and see if errors out? or is it covered already?

Reply:
This is an unsupported case and it has been covered by https://github.com/percona/percona-server/blob/5.7/mysql-test/extra/rpl_tests/rpl_extra_col_slave.test#L458-L802.

Reply:
Great! no need for any changes then